src/wireviz/wireviz.py

```diff
@@ -44,10 +45,11 @@ def parse(yaml_input: str, file_out: (str, Path) = None, return_types: (None, st
             if len(yaml_data[sec]) > 0:
                 if ty == dict:
                     for key, attribs in yaml_data[sec].items():
-                        # The Image dataclass might need to open an image file with a relative path.
-                        image = attribs.get('image')
-                        if isinstance(image, dict):
-                            image['gv_dir'] = Path(file_out if file_out else '').parent # Inject context
+                        if attribs.get('image'):
+                            image_path = attribs['image']['src']
+                            if not Path(image_path).is_absolute():  # resolve relative image path
+                                image_path = (Path(base_path) / image_path).resolve()
+                                attribs['image']['src'] = image_path
```

Comment:
I've already suggested an alternative to this code in https://github.com/formatc1702/WireViz/pull/189#discussion_r532944282. Why do you seem to prefer repeating lookup the `'image'` key 3 times instead of assigning and reusing an image variable?